### DISCUSSION TITLE
database-specific commands like SQLite .dump

### DISCUSSION THREAD

**Discussion by 0xBrian** [diesel-rs/diesel#2711] [Q&A] [answered]
Commands to SQLite like .dump, are these something that can be transmitted by diesel? In other words, can I get SQLite to create a database dump programmatically via diesel using some raw command interface?

**Comment by weiznich** [comment, marked as answer]
Commands like `.dump` are implemented as part of the sqlite3 command line tool and not as part of `libsqlite3`. Diesel depends on `libsqlite3`, not on the command line tool. That means diesel has no access to this functionality. 
That written: As the command line tool uses `libsqlite3` internally it should be possible to implement a similar functionality on top of that library (and on top of diesel).

**Reply by 0xBrian** [reply]
I see. Thank you!
